packages/knip/src/plugins/husky/index.ts

```diff
@@ -12,15 +13,35 @@ const ENABLERS = ['husky'];
 
 const isEnabled: IsPluginEnabledCallback = ({ dependencies }) => hasDependency(dependencies, ENABLERS);
 
-const gitHookPaths = getGitHookPaths('.husky');
+const gitHooksPathInV8 = getGitHookPaths('.husky', true);
+// husky v9 registers hooks in .husky/_/ to git and calls user defined hooks in .husky/ from there
+const gitHookPathsInV9 = getGitHookPaths('.husky', false);
 
-const CONFIG_FILE_PATTERNS = [...gitHookPaths];
+// Add patterns for both v8 and v9 because we can't know which version is installed at this point
```

Comment:
I understand the issue. This is OK for now.

Another solution we could think about is to split the plugin in two. Because the `isEnabled` function receives the `manifest` as well. So the "husky-v8" can be enabled separately from the "husky-v9" plugin.

But let's merge this plugin first, if you like this idea you could work on it later if you want.

Reply:
Thanks for the advice! I didn't notice that.

I worked on it but faced an issue with the plugin names.
If I change them, the configuration of the husky plugin won't work the same way as it does now.

I came up with the idea to make an alias `husky` for `husky-v9` here, but I'm not sure it would work without any problems.
https://github.com/webpro/knip/blob/957e02a1f6ef8d92f7240a5a3c9d6e6e360c9ef0/packages/knip/src/plugins/index.ts#L17

Shall I open another PR for this?

Reply:
Thanks for working on this!

> If I change them, the configuration of the husky plugin won't work the same way as it does now.

Good point!

It could be confusing for the user: how to name the config in `knip.json`, is it "husky8" or "husky-v8" or "husky"? Maybe it's not too bad that we handle the version differences in a single plugin (so we leave it as it is in `main` now). WDYT?

Reply:
I agree with the idea. Leave the single plugin as is might be the best way to avoid confusion.